src/features/users/usersSlice.js

```diff
@@ -0,0 +1,45 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  loading: false,
+  users: [
+    {
+      id: 1,
+      name: 'My profile',
+      description: 'User\'s description',
+      missionsByUser: [],
+      rocketsByUser: [],
+    },
+  ],
+  status: '',
+  error: '',
+};
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {
+    checkUser: (state, action) => {
+      state.users = action.payload === 'Under construction'
+        ? 'Under construction'
+        : state.users;
+    },
+  },
+
+});
+
+export const selectAllUsers = (state) => state.users.users;
+
+export const getUserStatus = (state) => state.users.status;
+
+export const getTotalMissionsByUser = (state) => state.users.totalMissionsByUser;
+
+export const getTotalRocketsByUser = (state) => state.users.totalRocketsByUser;
```

Comment:
> - [ ] As per the project requirements, please utilize the `filter()` function to render the list of all joined missions and reserved rockets. This approach is beneficial because it aligns with the given requirements and ensures the rendering of the desired lists in an efficient manner. 👍
> 
> ![ok](https://github.com/diarisdiakite/space-travelers/assets/40009354/d78e59e9-9954-4d7d-895f-637b3040d8d9)
> 
>

Reply:
We are using filter() to render the list of joined missions @Johnadibe. Please see the code in the slice. You may have noticed that these selectors const getTotalMissionsByUser and getTotalRocketsByUser are not the ones used in the code;
Here are the used selectors:
export const selectAllJoinedMissions = (state) => state.missions.missions.filter(
  (mission) => mission.reserved === true,
);
export const getAllReservedRockets = (state) => state.rockets.rockets.filter(
  (rocket) => rocket.reserved === true,
);